### PR TITLE
Trigger gluten tests when creating velox patch

### PR DESCRIPTION
**Pull request by zhouyuan**
This patch adds the gluten tests when creating Velox patches

### CODE REVIEW

**Comment by zhouyuan** [comment]
CC: @zhejiangxiaomai

File: .github/workflows/unittest.yml [hunk 1]
```diff
@@ -24,3 +24,95 @@ jobs:
...
+    runs-on: self-hosted
+    steps:
+      - name: Setup docker container
+        run: |
+          git clone https://github.com/oap-project/gluten && cd gluten
+          #git config --global alias.pr '!f() { git fetch -fu ${2:-origin} refs/pull/$1/head:pr/$1 && git checkout pr/$1; }; f'
+          BRANCH=$(echo ${GITHUB_REF#refs/heads/})
+          sed -i 's/oap-project/${GITHUB_REPOSITORY_OWNER}/g' ep/build-velox/src/get_velox.sh
+          sed -i 's/VELOX_BRANCH=main/VELOX_BRANCH=${BRANCH}/g' ep/build-velox/src/get_velox.sh
+          EXTRA_DOCKER_OPTIONS="--name velox-backend-ubuntu2004-test-$GITHUB_RUN_ID -e NUM_THREADS=30 --detach" \
```
Comment (jinchengchenghh) [inline .github/workflows/unittest.yml:38]: Can the `GITHUB_RUN_ID` be possible same with gluten test at the same time?
Do we have enough resources to do the test? Now we always add the unit test in gluten repo to test velox code

File: .github/workflows/unittest.yml [hunk 2]
```diff
@@ -24,3 +24,95 @@ jobs:
...
+          ./build_arrow.sh --build_tests=ON --build_benchmarks=ON && \
+          cd /opt/gluten/ep/build-velox/src && \
+          ./get_velox.sh && \
+          ./build_velox.sh'
+      - name: Build Gluten CPP library
+        run: |
+          docker exec velox-backend-ubuntu2004-test-$GITHUB_RUN_ID bash -c '
+          cd /opt/gluten/cpp && \
+          mkdir build && cd build && cmake -DBUILD_TESTS=ON -DBUILD_BENCHMARKS=ON .. && make -j'
+      - name: Build for Spark 3.2.0
```
Comment (jinchengchenghh) [inline .github/workflows/unittest.yml:60]: `Build for Spark 3.2.0` and `Build for Spark 3.2.1` is useless here

File: .github/workflows/unittest.yml (hunk 1 above)
Comment (zhouyuan) [reply .github/workflows/unittest.yml:37]: > Can the `GITHUB_RUN_ID` be possible same with gluten test at the same time?

the ID should be unique AFAIK - I'm reading below docs from GH:
https://docs.github.com/en/actions/learn-github-actions/variables


> Do we have enough resources to do the test? Now we always add the unit test in gluten repo to test velox code

currently there's only one runner - we could add more testing resources to check. Indeed this patch tries to save the time on creating the "empty patch" on gluten. I also found there are a few velox patches merged with out triggering gluten tests.

File: .github/workflows/unittest.yml (hunk 2 above)
Comment (zhouyuan) [reply .github/workflows/unittest.yml]: removed Spark320/321/330 builds

**Review by jinchengchenghh — APPROVED** [review]

**Comment by zhouyuan** [comment]
note: we decided to drop this approach and to keep the "mock patch" in gluten way

**Pull request closed without merging**